pkg/network/node/egressip.go

```diff
@@ -141,13 +160,76 @@ func getMarkForVNID(vnid, masqueradeBit uint32) string {
 	return fmt.Sprintf("0x%08x", vnid)
 }
 
+func (eip *egressIPWatcher) runIPAssignmentResync(stopCh <-chan struct{}) {
+	var addrChan chan netlink.AddrUpdate
+	addrSubscribe := func() error {
+		addrChan = make(chan netlink.AddrUpdate)
+		return netlink.AddrSubscribeWithOptions(addrChan, stopCh, netlink.AddrSubscribeOptions{
+			ErrorCallback: func(err error) {
+				klog.Errorf("Failed during AddrSubscribe callback: %v", err)
+			},
+		})
+	}
+
+	syncIPs := func() {
+		eip.tracker.Lock()
+		defer eip.tracker.Unlock()
+
+		eip.SyncIPAssignments()
+	}
+
+	addressSyncTimer := time.NewTicker(defaultIPAssignmentResyncInterval)
```

Comment:
We don't need the timer. Trust the netlink notifications. (If we can't trust them, then there's no reason to bother using that API at all...)

Reply:
I initially included the timer as a last resort but I am ok with relying on netlink events only